### PR TITLE
SASS-4497 - Implement Create/Update and Delete Override in income-tax…

### PR DESCRIPTION
**Pull request by georgievh**
…-state-benefits.

### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-4497](https://jira.tools.tax.service.gov.uk/browse/SASS-4497)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you run the smoke tests?
- [x]  Have you run the performance tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [ ]  Have you run the smoke tests? (where applicable)
- [x]  Have you run the performance tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/models/mongo/StateBenefitsUserData.scala [hunk 1]
```diff
@@ -31,13 +32,15 @@ case class StateBenefitsUserData(benefitType: String,
...
-                                 isPriorSubmission: Boolean,
+                                 benefitDataType: String,
                                  claim: Option[ClaimCYAModel],
                                  lastUpdated: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC)) {
 
-  lazy val isHmrcData: Boolean = claim.exists(_.isHmrcData)
-
-  lazy val isNewClaim: Boolean = claim.exists(_.benefitId.isEmpty)
+  lazy val isPriorSubmission: Boolean = claim.exists(_.benefitId.isDefined)
+  lazy val isNewClaim: Boolean = !isPriorSubmission
```
Comment (nickgravgaard) [inline app/models/mongo/StateBenefitsUserData.scala:40]: Do we need to have both `isPriorSubmission` and `isNewClaim`?

Comment (georgievh) [reply app/models/mongo/StateBenefitsUserData.scala:40]: If you are in favor to use positives when we do If clauses, than Yes :)

File: app/services/IntegrationFrameworkService.scala [hunk 2]
```diff
@@ -16,78 +16,94 @@
...
-    integrationFrameworkConnector.addStateBenefit(userData.taxYear, userData.nino, AddStateBenefit(userData.benefitType, claimData))
-  } else {
-    val benefitId = claimData.benefitId.get
-    integrationFrameworkConnector.updateStateBenefit(userData.taxYear, userData.nino, benefitId, UpdateStateBenefit(claimData)).map {
-      case Left(error) => Left(error)
+  private def createOrUpdateStateBenefit(userData: StateBenefitsUserData)
+                                        (implicit hc: HeaderCarrier): EitherT[Future, ApiServiceError, UUID] = userData match {
+    case _ if userData.isHmrcData => EitherT(Future.successful[Either[ApiServiceError, UUID]](Right(userData.claim.get.benefitId.get)))
+    case _ if userData.isNewClaim => addCustomerStateBenefit(userData)
+    case _ => updateCustomerStateBenefit(userData)
```
Comment (nickgravgaard) [inline app/services/IntegrationFrameworkService.scala:79]: Not sure how valuable the use of a match with guards is over regular if elses. I guess we use fewer lines...

Comment (georgievh) [reply app/services/IntegrationFrameworkService.scala:79]: Yes, and it is more readable

**Pull request merged**